### PR TITLE
Cloud Foundry Bosh Release

### PR DESCRIPTION
**Pull request by crobert-1**
This is the Bosh Release that can be used to deploy the Splunk OpenTelemetry Collector to a Pivotal Cloud Foundry (PCF) environment. This also includes a GitHub workflow for making sure the Bosh release can be successfully created whenever it's changed.

### CODE REVIEW

**Review by jvoravong — APPROVED** [review]
LGTM. You may want another review from someone else.

File: deployments/cloudfoundry/bosh/DEV_README.MD [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Development Guide
```
Comment (jeffreyc-splunk) [inline deployments/cloudfoundry/bosh/DEV_README.MD:1]: Maybe renaming this file to `DEVELOPMENT.md` would be more conventional?

File: deployments/cloudfoundry/bosh/config/blobs.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+signalfx_agent/signalfx-agent.tar.gz:
+  size: 194032255
+  sha: sha256:40ca24994f792d4b7c40f8640bea1fc420fbfe5eed918b4d63b4293e3b1e07dd
+splunk_otel_collector/splunk_otel_collector:
+  size: 163463346
+  sha: sha256:db0ddd022c2b8f4fa00d3d8f9764fcee4cc1de9fb2da32b1f7d536b677c89261
```
Comment (jeffreyc-splunk) [inline deployments/cloudfoundry/bosh/config/blobs.yml:6]: I'm not familiar with bosh.  Are these sizes and checksums used to validate the downloaded packages?  Does this file need to updated when the package versions are changed?

File: deployments/cloudfoundry/bosh/release [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+# This script is used to create a Bosh release of the Splunk distribution of the OpenTelemetry Collector.
+# This is intended to be run by the Pivotal Cloud Foundry (PCF) tile. The PCF tile should specify
+# proper variables as described below.
+
+# Variables:
```
Comment (jeffreyc-splunk) [inline deployments/cloudfoundry/bosh/release:10]: Should these env vars be documented in the dev guide and/or readme?

File: deployments/cloudfoundry/bosh/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# OpenTelemetry Collector BOSH Release
```
[the remark this answers is not in the document]
Comment (crobert-1) [reply deployments/cloudfoundry/bosh/README.md:1]: Fixed

File: deployments/cloudfoundry/bosh/DEV_README.MD (hunk 1 above)
Comment (crobert-1) [reply deployments/cloudfoundry/bosh/DEV_README.MD:1]: I was having a hard time deciding how to name it, that's much better. I'll update it, thanks!

File: deployments/cloudfoundry/bosh/packages/signalfx_agent/packaging [hunk 5]
```diff
@@ -0,0 +1,11 @@
+set -e -x
+
+tar xzf signalfx_agent/signalfx-agent.tar.gz
+
+cp -r signalfx-agent/* ${BOSH_INSTALL_TARGET}/
+
+pushd $BOSH_INSTALL_TARGET
+
+bin/patch-interpreter $(pwd)
```
[the remark this answers is not in the document]
Comment (crobert-1) [reply deployments/cloudfoundry/bosh/packages/signalfx_agent/packaging:9]: I've removed them now.

File: deployments/cloudfoundry/bosh/config/blobs.yml (hunk 2 above)
Comment (crobert-1) [reply deployments/cloudfoundry/bosh/config/blobs.yml:6]: Yes, they're sizes and checksums of the blobs (versioned files essentially in this case). These get updated by the `bosh add-blob` command, which is run by the release script, so there shouldn't be a need for the repo itself to be updated when package versions are changed.

File: deployments/cloudfoundry/bosh/release (hunk 3 above)
Comment (crobert-1) [reply deployments/cloudfoundry/bosh/release:10]: It's probably good to have a reference to them, but I don't want the two places to get out of sync. I'll add a comment where the release script is used in the REAMDE to reference the script itself for more variables.

File: deployments/cloudfoundry/bosh/DEVELOPMENT.MD [hunk 6]
```diff
@@ -0,0 +1,137 @@
+# Development Guide
```
Comment (jeffreyc-splunk) [inline deployments/cloudfoundry/bosh/DEVELOPMENT.MD:1]: From the latest link checker job: https://github.com/signalfx/splunk-otel-collector/runs/6119762127?check_suite_focus=true
```
[EXCLUDED] file:///github/workspace/deployments/cloudfoundry/bosh/DEVELOPMENT.MD | Excluded
```
Let's rename this file to `DEVELOPMENT.md` (lower-case `.md`).

**Review by jeffreyc-splunk — APPROVED** [review]

File: deployments/cloudfoundry/bosh/DEVELOPMENT.MD (hunk 6 above)
Comment (crobert-1) [reply deployments/cloudfoundry/bosh/DEVELOPMENT.MD]: Updated, thanks!

**Pull request merged**